src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -229,11 +248,10 @@ const AppFrame = ({ appUrl, allowedFeaturesList }: AppFrameProps): ReactElement
 
   useEffect(() => {
     const unsubscribe = txSubscribe(TxEvent.SAFE_APPS_REQUEST, async ({ safeAppRequestId, safeTxHash }) => {
-      const currentSafeAppRequestId = currentRequestId
-
-      if (currentSafeAppRequestId === safeAppRequestId) {
+      if (safeAppRequestId && currentRequestId === safeAppRequestId) {
         trackSafeAppEvent(SAFE_APPS_EVENTS.PROPOSE_TRANSACTION, appName)
         communicator?.send({ safeTxHash }, safeAppRequestId)
+        setTxFlow(undefined)
```

Comment:
Not related to this PR but this ensures that the tx modal closes when creating a normal safe apps tx.